system-variables.md

```diff
@@ -1,129 +1,836 @@
+
+### tidb_enable_cascades_planner
+
+- Scope: SESSION | GLOBAL
+- Default value: 0
+- This variable is used to control whether to enable the cascades planner feature.
+
+### tidb_enable_chunk_rpc <span class="version-mark">New in v4.0</span>
+
+- Scope: SESSION
+- Default value: 1
+- This variable is used to control whether to enable the `Chunk` data encoding format in Coprocessor.
+
+### tidb_enable_fast_analyze
+
+- Scope: SESSION | GLOBAL
+- Default value: 0, indicating not enabling the statistics fast `Analyze` feature.
+- This variable is used to set whether to enable the statistics `Fast Analyze` feature.
+- If the statistics `Fast Analyze` feature is enabled, TiDB randomly samples about 10,000 rows of data as statistics. When the data is distributed unevenly or the data size is small, the statistics accuracy is low. This might lead to a non-optimal execution plan, for example, selecting a wrong index. If the execution time of the regular `Analyze` statement is acceptable, it is recommended to disable the `Fast Analyze` feature.
+
+### tidb_enable_index_merge <span class="version-mark">New in v4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: 0
+- This variable is used to control whether to enable the index merge feature.
+
+### tidb_enable_noop_functions <span class="version-mark">New in v4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: 0
+- This variable is used to control whether to enable `get_lock` and `release_lock` functions. These two functions are not implemented, and always return 1 in the current version of TiDB.
+
+### tidb_enable_slow_log
+
+- Scope: SESSION
+- Default value: `1`
+- This variable is used to control whether to enable the slow log feature. It is enabled by default.
+
+### tidb_enable_stmt_summary <span class="version-mark">New in v3.0.4</span>
+
+- Scope: SESSION | GLOBAL
+- Default value: 1 (the value of the default configuration file)
+- This variable is used to control whether to enable the statement summary feature. If enabled, SQL execution information like time consumption is recorded to the `information_schema.STATEMENTS_SUMMARY` system table to identify and troubleshoot SQL performance issues.
+
+### tidb_enable_streaming
+
+- Scope: SERVER
+- Default value: 0
+- This variable is used to set whether to enable streaming.
+
+### tidb_enable_table_partition
+
+- Scope: SESSION | GLOBAL
+- Default value: "on"
+- This variable is used to set whether to enable the `TABLE PARTITION` feature.
+    - `off` indicates disabling the `TABLE PARTITION` feature. In this case, the syntax that creates a partition table can be executed, but the table created is not a partitioned one.
+    - `on` indicates enabling the `TABLE PARTITION` feature for the supported partition types. Currently, it indicates enabling range partition, hash partition and range column partition with one single column.
+    - `auto` functions the same way as `on` does.
```

Comment:
I recall that we have removed the `auto` value in TiDB 4.0.0, am I correct? @tiancaiamao

Reply:
It still seems to work when I set it:
```sql
mysql> show global variables like 'tidb_enable_table_partition';
+-----------------------------+-------+
| Variable_name               | Value |
+-----------------------------+-------+
| tidb_enable_table_partition | on    |
+-----------------------------+-------+
1 row in set (0.01 sec)

mysql> set global tidb_enable_table_partition='auto';
Query OK, 0 rows affected (0.02 sec)

mysql> show global variables like 'tidb_enable_table_partition';
+-----------------------------+-------+
| Variable_name               | Value |
+-----------------------------+-------+
| tidb_enable_table_partition | auto  |
+-----------------------------+-------+
1 row in set (0.00 sec)


```